src/components/button-group/__tests__/button-group.test.tsx

```diff
@@ -0,0 +1,53 @@
+import { render } from '@testing-library/react'
+import { ButtonGroup } from '../button-group'
+import { Button } from '@/components/button/button'
+
```

Comment:
**issue (non-blocking):** When writing test cases for a particular unit (the subject under test), it's helpful to apply the heuristic of _necessary_ and _sufficient_ to help determine if we're specifying the subject's behaviour well.

1. _necessary_ = tests should only test what they need to in order to fully exercise the subject;
2. _sufficient_ = tests should assert that the subject does everything a new implementation would also need to do.

With this heuristic in mind, we can assess the tests below and decide whether they meet the criteria of necessary and sufficient.

1. _Are they necessary?_ The first test is, but the latter two aren't, because they aren't specifying behaviour of the `ButtonGroup`. This is because the `ButtonGroup` does not care _what_ it renders, only that it renders whatever _children_ it has been given in a div _with the correct styles_.
2. _Are they sufficient?_ Kind of. Since we're only capturing a "DOM snapshot", not a visual snapshot, it's hard for us to sufficiently specify the gap between the `ButtonGroup`'s children is `--spacing-2` via an assertion. The best we can do is document that behaviour in the test description and otherwise explain why its not asserted by the test 🤷 

**suggestion:** I'd reduce this to a single test...
```javascript
test('renders its children horizontally with a gap of --spacing-2', () => {
  const { asFragment } = render(
    <ButtonGroup>
      <Button>Button 1</Button>
      <Button>Button 2</Button>
      <Button>Button 3</Button>
    </ButtonGroup>
  )
  // NOTE: We do not currently assert the children are laid out horizontally
  // with the correct gap because that behaviour is defined within CSS.
  expect(asFragment()).toMatchInlineSnapshot()
})
```

Reply:
Understood, I will update this.

Reply:
@kurtdoherty Thank you for your feedback.
**issue:** As elements prettier version is `3.2.5`, jest is throwing an error stating `Jest: Inline Snapshots are not supported when using Prettier 3.0.0 or above.`. 

**thought:** Though there are work around like version rollback to < `3.0.0`, I am thinking of using `toMatchSnapshot()` until prettier library is updated with a fix.

**question:** Do you have any other recommendation for this?

Reply:
Ah, that's a pain. All good, `toMatchSnapshot` is fine, it can just be nice to see them inline when they're nice and small.